src/components/fx/hover.js

```diff
@@ -625,6 +628,46 @@ function _hover(gd, evt, subplot, noHoverEvent) {
 
     hoverData.sort(function(d1, d2) { return d1.distance - d2.distance; });
 
+    // If in compare mode, select every point at position
+    if(hoverData[0].length !== 0 &&
+      ['x', 'y'].indexOf(mode) !== -1 &&
```

Comment:
concerning
```js
['x', 'y'].indexOf(mode) !== -1 
```
creating a dynamic array of strings just for index checking might be slow.
Could we create a constant e.g.
```js
var XY = ['x', 'y'];
```
at the upper scope and reuse that here?

Reply:
or even `var XY = {x: 1, y: 1}` -> `if(XY[mode])` 😅

Reply:
Done in https://github.com/plotly/plotly.js/pull/4664/commits/b8b9395f573a8b70e8263aac2ef154de49d6de5f